### ISSUE TITLE
Matrix operation

### ISSUE THREAD

**Issue by Aquarox** [Kitware/vtk-js#1232]
Hi !

I'm a bit confused of how I could manipulate an actor matrix.

For example, how am I suppose to reset the actormatrix to identity ?
I've seen there is a user matrix but I'm not sure how it suppose to work and I feel like it's not the best way to modify the actor matrix

Do you have any hints ?

Thanks !

**Comment by martinken** [comment]
setPosition(0,0,0)
setScale(1.0,1.0,1.0)

and setOrientation but it doesn't seem to be implemented. I recommend against using user matrix I'm surprised I didn't throw a fit when someone added it. Always leads to stuff being broken.

**Comment by jourdain** [comment]
FYI @finetjul

**Comment by Aquarox** [comment]
> 
> 
> setPosition(0,0,0)
> setScale(1.0,1.0,1.0)
> 
> and setOrientation but it doesn't seem to be implemented. I recommend against using user matrix I'm surprised I didn't throw a fit when someone added it. Always leads to stuff being broken.

Oh ok ! Thank you ! 

Do we have any idea when setOrientation would be implemented (that's exactly the one I need) ?

**Comment by finetjul** [comment]
You can set orientation through `rotateX`, `rotateY`, `rotateZ` and `rotateWXYZ()`.

@martinken would you suggest setUserMatrix() to internally set `position`, `scale` and `orientation` ? If not, what should we do when we only have a matrix ?

**Comment by bastiends** [comment]
Yeah I know about those functions but using rotateX, rotateY and rotateZ, I can't correctly combine rotations

**Comment by martinken** [comment]
Yes, SetUserMatrix should be more like SetPoseFromMatrix() that sets position scale and orientation from a a user provided matrix. Same idea in Camera. The problem with user matrix both in a prop and in the camera is that there is code that may look at position/clipping range/parallel for example. When a user matrix is set those methods may not return the correct value. It just returns the ivar part which would have to be run through the user matrix to get the real value. How does getparallel() work when the camera has a user matrix? It is almost certainly broken right now for that case. It could be made to work, but right now that isn't how it is coded.
